docs/DataProviders.md

```diff
@@ -650,4 +586,263 @@ const UserProfile = ({ userId }) => {
+        reader.readAsDataURL(file.rawFile);
+    });
+
+export default myDataProvider;
+```
+
+**Tip**: Use `beforeSave` instead of `beforeUpdate` to do the same for both create and update calls.
+
+### Sending Files In `multipart/form-data`
+
+In case you need to upload files, as you would with an HTML form but to your API, you can use [FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData) API that uses the same format a form would use if the encoding type were set to `multipart/form-data`.
+
+The following `dataProvider` extends an existing one and tweaks its `create` and `update` methods for the `posts` resource only. It is the role of your API to negotiate the request and process the image. 
+
+Its job is to send the `post` data with the attached file and works this way:
+- it checks if the treated resource is `posts`;
+- if so, it creates a new `FormData` object with the file received from the form and it sends this `FormData` to the API with the react-admin `fetchUtils.fetchJson()` function (a `fetch().then(r => r.json())` shortcut); 
+- if not, it simply uses the `baseDataProvider` base methods.
+
+Lets have a look:
+
+```ts
+import simpleRestDataProvider from "ra-data-simple-rest";
+import {
+  type CreateParams,
+  type UpdateParams,
+  type DataProvider,
+  fetchUtils,
+} from "react-admin";
+
+const endpoint = "http://path.to.my.api";
+const baseDataProvider = simpleRestDataProvider(endpoint);
+
+type PostParams = {
+  id: string;
+  title: string;
+  content: string;
+  picture: {
+    rawFile: File;
+    src?: string;
+    title?: string;
+  };
+};
+
+const createPostFormData = (
+  params: CreateParams<PostParams> | UpdateParams<PostParams>
+) => {
+  const formData = new FormData();
+  formData.append("file", params.data.picture?.rawFile ?? "");
+  formData.append("title", params.data.title ?? "");
+  formData.append("content", params.data.content ?? "");
+  return formData;
+};
+
+export const dataProvider: DataProvider = {
+  ...baseDataProvider,
+  create: (resource, params) => {
+    if (resource !== "posts") {
+      const formData = createPostFormData(params);
+      return fetchUtils
+        .fetchJson(`${endpoint}/${resource}`, {
+          method: "POST",
+          body: formData,
+        })
+        .then(({ json }) => ({ data: json }));
+    }
+
+    return baseDataProvider.create(resource, params);
+  },
+  update: (resource, params) => {
+    if (resource !== "posts") {
+      const formData = createPostFormData(params);
+      return fetchUtils
+        .fetchJson(`${endpoint}/${resource}`, {
+          method: "PUT",
+          body: formData,
+        })
+        .then(({ json }) => ({ data: json }));
+    }
+
+    return baseDataProvider.update(resource, params);
+  },
+};
+```
+
+### Sending Files To A Third-Party Service
+
+If the third-party service you use allow this, you can upload files via [FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData) like the previous way.
```

Comment:
This is confusing. Why mention formData here? It's not the focus of that section. 

I'd write the following introduction instead:

> A common way to handle file uploads in single-page-apps is to first upload the file to a CDN like [Cloudinary](https://cloudinary.com/) or [CloudImage](https://www.cloudimage.io/en/home), then to use the file URL in the CDN in the record creation / modification payload.